sympy/series/sequences.py

```diff
@@ -0,0 +1,1123 @@
+
+    def __add__(self, other):
+        return self.add(other)
+
+    def __neg__(self):
+        return self.coeff_mul(-1)
+
+    def __sub__(self, other):
+        return self.sub(other)
+
+    def __mul__(self, other):
+        return self.mul(other)
+
+    def __iter__(self):
+        i = 0
+        while(i < self.length):
+            pt = self._ith_point(i)
+            yield self.coeff(pt)
+            i += 1
+
+    def __getitem__(self, index):
+        if isinstance(index, integer_types):
+            index = self._ith_point(index)
+            return self.coeff(index)
+        elif isinstance(index, slice):
+            start, stop = index.start, index.stop
+            if start is None:
+                start = 0
+            if stop is None:
+                stop = self.length
+            return [self.coeff(self._ith_point(i, index.step)) for i in\
+                               range(start, stop)]
+
+
+class EmptySequence(with_metaclass(Singleton, SeqBase)):
+    """
+    Represents an empty sequence. The empty sequence is available as a
+    singleton as S.EmptySequence.
+
+    Examples
+    ========
+
+    >>> from sympy import S, SeqPer, oo
+    >>> S.EmptySequence
+    EmptySequence()
+
+    >>> SeqPer((1, 2)).add(S.EmptySequence)
+    SeqPer((1, 2), ([0, oo), 1))
+    >>> SeqPer((1, 2)).mul(S.EmptySequence)
+    EmptySequence()
+    >>> S.EmptySequence.coeff_mul(-1)
+    EmptySequence()
+    """
+
+    @property
+    def interval(self):
+        return S.EmptySet
+
+    @property
+    def length(self):
+        return S.Zero
+
+    def coeff_mul(self, coeff):
+        """See docstring of SeqBase.coeff_mul"""
+        return self
+
+    def __iter__(self):
+        return iter([])
+
+
+def _parse_interval(interval):
+    """
+    interval should be of the form (start, step) or (start, stop, step)
+    Both start and stop cannot be unbounded
+    step cannot be unbounded
+
+    Allowed:
+    * Any instance of set
+    * (Set, step)
+    * (start, stop)
+    * (start, stop, step)
+
+    returns an Interval object and a step value
+
+    Examples
+    ========
+
+    >>> from sympy.series.sequences import _parse_interval as pari
+    >>> from sympy import Interval
+    >>> pari(Interval(0, 5))
+    ([0, 5], 1)
+    >>> pari((Interval(0, 5), 2))
+    ([0, 5], 2)
+    >>> pari((0, 5))
+    ([0, 5], 1)
+    >>> pari((0, 5, 2))
+    ([0, 5], 2)
+    """
+    start, stop, step = None, None, None
+    if isinstance(interval, Set):
+        start, stop = interval.inf, interval.sup
+    elif is_sequence(interval, Tuple):
+        if len(interval) == 2:
+            if isinstance(interval[0], Set):
+                start, stop = interval[0].inf, interval[0].sup
+                step = interval[1]
+            else:
+                start, stop = interval
+        elif len(interval) == 3:
+            start, stop, step = interval
+
+    if step == None:
+        step = 1 # default
+
+    if start == None or stop == None:
```

Comment:
Same here. I'm not going to tag all of them, please go through and cleanup these in all files. There's a few other places.